binders/kafka-binder/spring-cloud-stream-binder-kafka-core/src/main/java/org/springframework/cloud/stream/binder/kafka/properties/KafkaBinderConfigurationProperties.java

```diff
@@ -184,41 +186,35 @@ public String getKafkaConnectionString() {
 		try {
-			moveBrokerCertsIfApplicable();
-			moveSchemaRegistryCertsIfApplicable();
+			// broker certificates
+			moveCertsIfApplicable("ssl.truststore.location");
+			moveCertsIfApplicable("ssl.keystore.location");
+
+			// schema registry certificates
+			moveCertsIfApplicable("schema.registry.ssl.truststore.location");
+			moveCertsIfApplicable("schema.registry.ssl.keystore.location");
 		}
 		catch (Exception e) {
 			throw new IllegalStateException(e);
 		}
 	}
 
-	private void moveBrokerCertsIfApplicable() throws IOException {
-		final String trustStoreLocation = this.configuration.get("ssl.truststore.location");
-		if (trustStoreLocation != null && trustStoreLocation.startsWith("classpath:")) {
-			final String fileSystemLocation = moveCertToFileSystem(trustStoreLocation, this.certificateStoreDirectory);
-			// Overriding the value with absolute filesystem path.
-			this.configuration.put("ssl.truststore.location", fileSystemLocation);
-		}
-		final String keyStoreLocation = this.configuration.get("ssl.keystore.location");
-		if (keyStoreLocation != null && keyStoreLocation.startsWith("classpath:")) {
-			final String fileSystemLocation = moveCertToFileSystem(keyStoreLocation, this.certificateStoreDirectory);
+	private void moveCertsIfApplicable(String storeProperty) throws IOException {
+		final String storeLocation = this.configuration.get(storeProperty);
+
+		// If the path is not defined, or it is a local file path do not move the file
+		if (storeLocation != null && !checkIfFileExists(storeLocation)) {
+			final String fileSystemLocation = moveCertToFileSystem(storeLocation, this.certificateStoreDirectory);
 			// Overriding the value with absolute filesystem path.
-			this.configuration.put("ssl.keystore.location", fileSystemLocation);
+			this.configuration.put(storeProperty, fileSystemLocation);
 		}
 	}
 
-	private void moveSchemaRegistryCertsIfApplicable() throws IOException {
-		String trustStoreLocation = this.configuration.get("schema.registry.ssl.truststore.location");
-		if (trustStoreLocation != null && trustStoreLocation.startsWith("classpath:")) {
-			final String fileSystemLocation = moveCertToFileSystem(trustStoreLocation, this.certificateStoreDirectory);
-			// Overriding the value with absolute filesystem path.
-			this.configuration.put("schema.registry.ssl.truststore.location", fileSystemLocation);
-		}
-		final String keyStoreLocation = this.configuration.get("schema.registry.ssl.keystore.location");
-		if (keyStoreLocation != null && keyStoreLocation.startsWith("classpath:")) {
-			final String fileSystemLocation = moveCertToFileSystem(keyStoreLocation, this.certificateStoreDirectory);
-			// Overriding the value with absolute filesystem path.
-			this.configuration.put("schema.registry.ssl.keystore.location", fileSystemLocation);
+	private boolean checkIfFileExists(String path) {
+		try {
+			return Files.isRegularFile(Paths.get(path));
+		} catch (Exception e) {
```

Comment:
This needs to be on a new line. Can you run a full build and see if there are any checkstyle failures?